### ISSUE TITLE
portus + thin, please provide detailed instructions

### ISSUE THREAD

**Issue by aleksyst** [SUSE/Portus#901]
I'm using portus with private docker registry on OpenSuse Leap.
Unfortunately I've got an issue with portus hanging on almost every docker push.
I was trying to follow Your tutorial on portus + nginx + thin from http://port.us.org/docs/setups/3_nginx_bare_metal.html
I'm not familiar with ruby, can You please explain in detail how exactly do You setup thin to work with Portus ?
I managed to install thin gem and all depenancies. What should I do next ?

**Comment by mssola** [comment]
If you are using openSUSE Leap, you should probably take a look at the [rpm](http://port.us.org/docs/setups/1_rpm_packages.html). Updates on the stable 2.0.x release are happening in [this](https://build.opensuse.org/project/show/Virtualization:containers:Portus:2.0) repository. Thus, a simple:

    $ zypper ar http://download.opensuse.org/repositories/Virtualization:/containers:/Portus:/2.0/openSUSE_Leap_42.1/
    $ zypper in portus

Should give you Portus 2.0.5, which is the current stable version. As you can read in the [documentation of the RPM](http://port.us.org/docs/setups/1_rpm_packages.html), then you can just configure Portus in the way it's best for your case.

That being said, if you prefer to go in the NGinx + Thin path, then it should be enough to just raise the number of `worker_processes` in your NGinx config, or the `servers` of Thin, as explained [here](http://port.us.org/docs/setups/3_nginx_bare_metal.html#thin). I'm personally using this same setup but with Puma instead of Thin, and it's working flawlessly. If you are still facing problems with Thin maybe you want to give Puma a try :wink:

**Comment by aleksyst** [comment]
Not really helpfull, but probably you expected some level of ruby knowledge. Pity I have none...
I cant get porus to work with thin, because I  dont know how to do that.
I guess I'm stuck with restarting apache on every docker posh ....

**Comment by mssola** [comment]
@aleksyst if you don't have experience configuring a ruby application, and you are on openSUSE Leap, I *really* recommend using the RPM provided on OBS.

**Comment by aleksyst** [comment]
I've got it working with thin eventually after much trial and error.
Here's rough overview of what I've done:

Installed nginx enabled it and disabled apache2

Got sample configs from http://port.us.org/docs/setups/3_nginx_bare_metal.html and adjusted them to my environment.

Installed needed packages: ruby-devel gcc make nodejs mysql-devel gcc-c++

Installed needed gems: mysql2 -v '0.3.18' bundler thin execjs

Added gems to portus Gemfile: thin and execjs

Added socket folder in /srv/Portus/tmp and gave permissions to nginx user

Did bundle install

Adjusted start command to supply missing parameters:
bundle exec thin start -C config/thin.yml -A rack -e production

Executed command from Portus directory

Puma didn't work in my case, while registry worked correctly, website refuse to load throwing css assets errors.
